### PR TITLE
Add vk_function_loader.hpp

### PR DESCRIPTION
**Pull request by charles-lunarg**
Adds a new header only utility library which contains a loader for the Vulkan-Loader,
tables to hold vulkan function pointers, and init functions to fill them out.

This is for #73 - will close the issue when VVL & VEL are using this library.

### CODE REVIEW

File: tests/test_setting_env.cpp [hunk 1]
```diff
@@ -439,4 +439,6 @@ TEST(test_layer_setting_env, vkuGetLayerSettingValues_String) {
     EXPECT_EQ(2, value_count);
 
     vkuDestroyLayerSettingSet(layerSettingSet, nullptr);
+
+    SetEnv("VK_LUNARG_TEST_MY_SETTING=");
```
Comment (juan-lunarg) [inline tests/test_setting_env.cpp:443]: Make a separate PR so we can just merge this fix in.

**Review by juan-lunarg — CHANGES_REQUESTED** [review]
I'd like to see a branch with the functionality implemented in VEL or VVL.

File: tests/test_setting_env.cpp (hunk 1 above)
Comment (charles-lunarg) [reply tests/test_setting_env.cpp:443]: Done
(doing some more changes before pushing).

**Comment by charles-lunarg** [comment]
Note that this PR relies on #155 so I will rebase when that is merged.

**Comment by charles-lunarg** [comment]
Opened this up a month ago, but the reality is that the solution is to use volk, not write another new library. If there is future interest in library for function loading in this repo, it can be revisited.
